### DISCUSSION TITLE
Lit component library or design system setup

### DISCUSSION THREAD

**Discussion by lan-chan** [lit/lit#3550] [Q&A] [answered]
I have searched the lit.dev and open-wc.org documentation but cannot seem to find examples of how to create an npm package that exports multiple lit web-components.

Can anyone point me to some example libraries that do this?
Thanks!

**Comment by lkraav** [comment]
See https://github.com/conversionxl/aybolit/blob/master/packages/cxl-ui/src/index.js (and its imports) for an example.

**Reply by lan-chan** [reply]
Thanks for the input. Looks like i'll have to check this `pika` thing out.

**Comment by justinfagnani** [comment, marked as answer]
All you need is multiple file with web components in them. If you take the Lit starter kit, which has a single my-element.js file, you can add a my-element2.js with another element.

Users can import:

```ts
import 'my-package/my-element.js';
// and
import 'my-package/my-element2.js';
```

**Reply by lan-chan** [reply]
I've tried this and it works - thanks.
I'm not a fan of how the build process just drops the built files in the root dir though.

To fix this I updated the `tsconfig.json` with:
```json
"compilerOptions": {
  "outDir": "build"
}
```
Then added exports to the `package.json`
```json
"exports": {
  "./my-element": "./build/my-element.js",
  "./my-element-2": "./build/my-element-2.js"
}
```

**Reply by justinfagnani** [reply]
Yes, that's fine too

Just know that client that do not expect package exports will have to import with `my-package/build/my-element.js` still. If you're sure that all your clients support package exports, then you can have the files system layout and the import layout be different like that.
